scp/scpEraseOperatorInterpreter.cpp

```diff
@@ -12,63 +12,103 @@
 #include "erase_operators/SCPOperatorEraseElStr5.hpp"
 #include "erase_operators/SCPOperatorEraseSetStr3.hpp"
 #include "erase_operators/SCPOperatorEraseSetStr5.hpp"
-#include "sc-memory/sc_memory.hpp"
+
 #include <iostream>
 
-namespace scp {
-ScAddr ASCPEraseOperatorInterpreter::msAgentKeynode;
+namespace scp
+{
+ScAddrToValueUnorderedMap<std::function<SCPOperator *(ScMemoryContext &, ScAddr)>>
+    ASCPEraseOperatorInterpreter::supportedOperators = {};
 
-SC_AGENT_IMPLEMENTATION(ASCPEraseOperatorInterpreter)
+ScResult ASCPEraseOperatorInterpreter::DoProgram(
+    ScEventAfterGenerateOutgoingArc<ScType::EdgeAccessConstPosPerm> const & event,
+    ScAction & action)
 {
-    if (!edgeAddr.IsValid())
-        return SC_RESULT_ERROR;
-
-    ScAddr scp_operator =m_memoryCtx.GetEdgeTarget(edgeAddr);
-
-    ScAddr type;
-    if (SC_TRUE != Utils::resolveOperatorType(m_memoryCtx, scp_operator, type))
-        return SC_RESULT_ERROR_INVALID_TYPE;
-
-    SCPOperator* oper = nullptr;
-    if (type == Keynodes::op_eraseEl)
-    {
-        oper = new SCPOperatorEraseEl(m_memoryCtx, scp_operator);
-    }
-    if (type == Keynodes::op_eraseElStr3)
-    {
-        oper = new SCPOperatorEraseElStr3(m_memoryCtx, scp_operator);
-    }
-    if (type == Keynodes::op_eraseElStr5)
-    {
-        oper = new SCPOperatorEraseElStr5(m_memoryCtx, scp_operator);
-    }
-    if (type == Keynodes::op_eraseSetStr3)
-    {
-        oper = new SCPOperatorEraseSetStr3(m_memoryCtx, scp_operator);
-    }
-    if (type == Keynodes::op_eraseSetStr5)
-    {
-        oper = new SCPOperatorEraseSetStr5(m_memoryCtx, scp_operator);
-    }
-    if (oper == nullptr)
-        return SC_RESULT_ERROR_INVALID_PARAMS;
+  if (!event.GetArc().IsValid())
+    return action.FinishUnsuccessfully();
+
+  ScAddr scp_operator = event.GetOtherElement();
+
+  ScAddr type;
+  if (!Utils::resolveOperatorType(m_context, scp_operator, type))
```

Comment:
```suggestion
  if (!Utils::ResolveOperatorType(m_context, scp_operator, type))
```

Reply:
I didn't change utils file, there is too many methods in it, I'll leave it to an issue